### ISSUE TITLE
read_timeout not working

### ISSUE THREAD

**Issue by wf5w** [hparra/ruby-serialport#73]
I am running with: ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu] on Linux Mint 20.3 

I have the following code snippet, but here is what I would expect: I would expect that after say 250 ms, that if there was no input then an error would be produced. EIO? but if a character came then the loop would continue. What is actually happening is nothing, it just hangs:

```sp = SerialPort.new(port, baud, 8, 1, SerialPort::NONE)  # port = /dev/pts/1 (for testing), baud = 115200
bytes = []

begin
  sp.read_timeout = 250
  puts "read timeout is set to: #{sp.read_timeout}\n"

  while 1
   bytes << sp.getc
   print bytes.last
    if bytes.last == 0xFD
      break
    end
  end
rescue => e
  unless e.class == Errno::EIO
    puts error.inspect
  end
end

puts "\n\nreceived: #{bytes.join('')}"
```

**Comment by sumitbirla** [comment]
I am also having this problem.   Were you able to solve this?

**Comment by wf5w** [comment]
No. And I haven't seen anything that says they even acknowledge there's a problem.--Sent from my Android phone with GMX Mail. Please excuse my brevity.On 10/24/22, 6:01 PM Sumit Birla ***@***.***> wrote:

   
  I am also having this problem. Were you able to solve this?

**Comment by wf5w** [comment]
Well ok then,

I would say this gem is Useless for anyone doing serious work.
on to python I guess or C

**Comment by tenderlove** [comment]
Please try [this gem](https://github.com/tenderlove/uart). It's meant as a replacement. You can use regular Ruby IO method calls like `wait_readable` (which takes a timeout).
